src/core/services.py

```
@@ -81,58 +76,135 @@ def run(
-                name=str(row.get(config.COL_NAME, "")),
-                scores={
-                    str(k): float(v)
-                    for k, v in row.items()
-                    if str(k).startswith(config.SCORE_PREFIX)
-                },
-                groupers=str(row.get(config.COL_GROUPER, "")),
-                separators=str(row.get(config.COL_SEPARATOR, "")),
-                original_index=i,
+        try:
+            if participants_df is None:
+                raise ValueError("Participants DataFrame cannot be None.")
```

Ruff (0.15.12)

[warning] Abstract `raise` to an inner function

(TRY301)